### PR TITLE
Updating to support latest Provider functions

### PR DESCRIPTION
**Pull request by joeharlan**
# Description

This PR includes some changes in formatting for readability, but also include some functional changes.  The main functional change includes code to support handling any empty role templates returned by the RSC API that the Azure RM provider interprets as a change on every plan/apply operation despite there being no actual changes.

## Related Issue

This project only accepts pull requests related to open issues.

* If suggesting a new feature or change, please discuss it in an issue first.
* If fixing a bug, there should be an issue describing it with steps to reproduce

[_Please link to the issue here_](https://github.com/rubrikinc/terraform-provider-polaris/issues/185)

## Motivation and Context

The proposed changes solve the previously mentioned issue with the RG-scoped Azure SQL DB and Azure SQL MI role templates containing no permissions.

## How Has This Been Tested?

* Please describe in detail how you tested your changes.
* Include details of your testing environment, and the tests you ran to see how your change affects other areas of the code, etc.

The changes were tested in the Oasis Labs environment using the 0.9.0-beta.8 polaris Provider.

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

Go over all the following points, and put an `x` in all the boxes that apply. If you're unsure about any of these, don't hesitate to ask. We're here to help!
- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the CHANGELOG file accordingly for the version that this merge modifies.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by DamaniN — CHANGES_REQUESTED** [review]
@joeharlan, the `"RubrikBackups-RG-DontDelete-terraform` Resource Group name is based on what RSC uses. Changing this here will be a breaking change for any existing customer who is using this module.  All resources in the RG will have to be removed and recreated, including backups. Alternatively, this RG name will need to be renamed in the customer's existing Terraform state file.

If you really want to make this change, can you include documentation about the breaking change? Also, please document how to rename the RGs in the existing state file before an update is run?

**Comment by joeharlan** [comment]
@DamaniN I am not against changing that back, but would be shocked to learn that anyone is using the module exanple direct from GitHub rather than cloning and modifying it to their specific needs and desires.  The change in question is also only in the README.md file and should not actually impact anyone who is using the module example in that way.

File: main.tf [hunk 1]
```diff
@@ -25,8 +28,9 @@ resource "azurerm_resource_group" "default" {
 # permissions.
 resource "azurerm_role_definition" "subscription" {
   for_each = toset(var.rsc_azure_features)
-  name        = "Rubrik Polaris SubRole ${each.key} - terraform - ${data.azurerm_subscription.current.subscription_id}"
-  description = "Rubrik Polaris Subscription role for ${each.key} - Terraform Generated"
+  
+  name        = "Rubrik Security Cloud SubRole ${each.key} - terraform - ${data.azurerm_subscription.current.subscription_id}"
+  description = "Rubrik Security Cloud Subscription role for ${each.key} - Terraform Generated"
```
Comment (DamaniN) [inline main.tf:33]: @joeharlan the name and description provided here match what RSC currently creates during the with and without OAuth workflows. Please confirm that you want to move away from this standard.

File: main.tf [hunk 2]
```diff
@@ -41,22 +45,34 @@ resource "azurerm_role_definition" "subscription" {
 # permissions.
 resource "azurerm_role_definition" "resource_group" {
   for_each = toset(var.rsc_azure_features)
-  name        = "Rubrik Polaris RGRole ${each.key} - terraform - ${data.azurerm_subscription.current.subscription_id}"
-  description = "Rubrik Polaris Resource Group role for ${each.key} - Terraform Generated"
+
+  name        = "Rubrik Security Cloud RGRole ${each.key} - terraform - ${data.azurerm_subscription.current.subscription_id}"
+  description = "Rubrik Security Cloud Resource Group role for ${each.key} - Terraform Generated"
```
Comment (DamaniN) [inline main.tf:50]: @joeharlan the name and description provided here match what RSC currently creates during the with and without OAuth workflows. Please confirm that you want to move away from this standard.

**Review by DamaniN — CHANGES_REQUESTED** [review]
@joeharlan the names and descriptions provided for the roles match what RSC currently creates during the with and without OAuth workflows. Please confirm that you want to move away from this standard. 

The changes you propose do make sense to remove the Polaris name, however, TF will now do something different than the UI. This may lead to confusion with support.

File: main.tf (hunk 1 above)
Comment (joeharlan) [reply main.tf:33]: @DamaniN The way I considered this versus how the "with OAuth" process creates the roles is they are mutually exclusive.  With the customers I have using these modules, each have their own naming standards.  These names were changed to match their preferences in every case.  This experience has led me to believe customers see these modules as prescriptive examples and not hard and fast rules.  Not one of them has used them as-is.  Instead, they have made significant modifications to suit their naming and IaC standards.  So long as they do not break things, this seems reasonable to permit.  If the customer is able to articulate which of their custom-named roles aligns to each CNP feature, then they should be able to effecively communucate with Rubrik Support.  Nothing about their names is hard-coded in the product from a functional standpoint.

File: main.tf (hunk 2 above)
Comment (joeharlan) [reply main.tf:50]: @DamaniN Same comment in this area as made in the other case you called out.  The names should be flexible to customer standards, so changes here to remove Polaris are compatible and immaterial.

**Review by DamaniN — APPROVED** [review]
LGTM

**Pull request merged**